pyseir/backtest/backtest.py

```diff
@@ -0,0 +1,432 @@
+import os
+import us
+import matplotlib.backends.backend_pdf
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from datetime import datetime, timedelta
+from matplotlib.dates import DateFormatter
+from multiprocessing import Pool
+from functools import partial
+from pyseir import load_data
+from pyseir.load_data import HospitalizationDataType
+from pyseir.inference.model_fitter import ModelFitter
+from pyseir.utils import get_run_artifact_path, RunArtifact
+from pyseir.backtest.timeseries_metrics import TimeSeriesMetrics, error_type_to_meaning
+
+
+REF_DATE = datetime(year=2020, month=1, day=1)
+
+def load_observations(fips, ref_date=datetime(year=2020, month=1, day=1)):
+    """
+    Load observations (new cases, new deaths and hospitalizations) for
+    given fips code.
+
+    Parameters
+    ----------
+    fips: str
+        FIPS code.
+    ref_date: Datetime
+        Reference start date.
+
+    Returns
+    -------
+    observations: pd.DataFrame
+        Contains observations for given fips codes, with columns:
+        - new_cases: float, observed new cases
+        - new_deaths: float, observed new deaths
+        - current_hosp: float, observed hospitalizatons
+        and dates of observation as index.
+    """
+
+    observations = {}
+    if len(fips) == 5:
+        times, observations['new_cases'], observations['new_deaths'] = \
+            load_data.load_new_case_data_by_fips(fips, ref_date)
+        hospital_times, hospitalizations, hospitalization_data_type = \
+            load_data.load_hospitalization_data(fips, t0=ref_date)
+        observations['times'] = times.values
+    elif len(fips) == 2:
+        state_obj = us.states.lookup(fips)
+        observations['times'], observations['new_cases'], observations['new_deaths'] = \
+            load_data.load_new_case_data_by_state(state_obj.name, ref_date)
+        hospital_times, hospitalizations, hospitalization_data_type = \
+            load_data.load_hospitalization_data_by_state(state_obj.abbr, t0=ref_date)
+        observations['times'] = np.array(observations['times'])
+
+    if hospitalization_data_type is HospitalizationDataType.CUMULATIVE_HOSPITALIZATIONS:
+        observations['current_hosp'] = np.full(observations['times'].shape[0], np.nan)
+        observations['current_hosp'][hospital_times - observations['times'].min()] = np.diff(hospitalizations)
```

Comment:
Current hospitalizations != new hospitalizations here.   What I have been doing is to instead try to reconstruct new hospitalizations from current by diffing, and then adding back in the expected discharge outflow. You can see an example in infer_rt. 

Also, we might as well move this method into load_data and I can migrate some of the other classes on to this so we don't have to repeat it all over.

Reply:
Ah, got it, let me see if I can add this to load_data.